Mutagen.Bethesda.Analyzers.Skyrim/Record/Armor/RaceAddonsAnalyzer.cs

```diff
@@ -0,0 +1,75 @@
+
+    public IEnumerable<TopicDefinition> Topics => [ArmorMissingRaceAddons];
+
+    private static readonly List<FormKey> DefaultPlayerRaces =
+    [
+        FormKeys.SkyrimSE.Skyrim.Race.ArgonianRace.FormKey, FormKeys.SkyrimSE.Skyrim.Race.ArgonianRaceVampire.FormKey,
+        FormKeys.SkyrimSE.Skyrim.Race.BretonRace.FormKey, FormKeys.SkyrimSE.Skyrim.Race.BretonRaceVampire.FormKey,
+        FormKeys.SkyrimSE.Skyrim.Race.DarkElfRace.FormKey, FormKeys.SkyrimSE.Skyrim.Race.DarkElfRaceVampire.FormKey,
+        FormKeys.SkyrimSE.Skyrim.Race.HighElfRace.FormKey, FormKeys.SkyrimSE.Skyrim.Race.HighElfRaceVampire.FormKey,
+        FormKeys.SkyrimSE.Skyrim.Race.ImperialRace.FormKey, FormKeys.SkyrimSE.Skyrim.Race.ImperialRaceVampire.FormKey,
+        FormKeys.SkyrimSE.Skyrim.Race.KhajiitRace.FormKey, FormKeys.SkyrimSE.Skyrim.Race.KhajiitRaceVampire.FormKey,
+        FormKeys.SkyrimSE.Skyrim.Race.NordRace.FormKey, FormKeys.SkyrimSE.Skyrim.Race.NordRaceVampire.FormKey,
+        FormKeys.SkyrimSE.Skyrim.Race.OrcRace.FormKey, FormKeys.SkyrimSE.Skyrim.Race.OrcRaceVampire.FormKey,
+        FormKeys.SkyrimSE.Skyrim.Race.RedguardRace.FormKey, FormKeys.SkyrimSE.Skyrim.Race.RedguardRaceVampire.FormKey,
+        FormKeys.SkyrimSE.Skyrim.Race.WoodElfRace.FormKey, FormKeys.SkyrimSE.Skyrim.Race.WoodElfRaceVampire.FormKey
+    ];
+
+    public RecordAnalyzerResult? AnalyzeRecord(ContextualRecordAnalyzerParams<IArmorGetter> param)
+    {
+        var armor = param.Record;
+
+        // Armor with template armor inherit all relevant data from the template armor and should not be checked themselves
+        if (!armor.TemplateArmor.IsNull) return null;
+
+        // Non-playable cannot be equipped by the player and is usually required to be universally compatible
+        if (armor.MajorFlags.HasFlag(Bethesda.Skyrim.Armor.MajorFlag.NonPlayable)) return null;
+
+        // Exclude non player race armor
+        if (armor.Race.FormKey != FormKeys.SkyrimSE.Skyrim.Race.DefaultRace.FormKey
+            && !DefaultPlayerRaces.Contains(armor.Race.FormKey)) return null;
+
+        // Build list of all races in armor addons
+        var missingRaces = new List<FormKey>(DefaultPlayerRaces);
+        foreach (var armorAddon in armor.Armature)
+        {
+            var addon = armorAddon.TryResolve(param.LinkCache);
+            if (addon == null) continue;
+
+            foreach (var additionalRace in addon.AdditionalRaces)
+            {
+                missingRaces.Remove(additionalRace.FormKey);
+            }
+            missingRaces.Remove(addon.Race.FormKey);
+        }
+
+        var result = new RecordAnalyzerResult();
+
+        foreach (var race in missingRaces)
+        {
+            if (param.LinkCache.TryResolve<IRaceGetter>(race, out var r))
+            {
+                result.AddTopic(RecordTopic.Create(
+                    armor,
+                    ArmorMissingRaceAddons.Format(r.EditorID),
+                    x => x.EditorID));
```

Comment:
I think `x => x.EditorID` is a misuse of this input.

The purpose of `memberExpression` when making a topic is to inform the engine what member of the original record of interest it should watch is.

For example.  In a simpler analyzer, like say one that makes sure Npc weights are above zero, or something... it would put `x => x.Weight`
- Engine runs analyzer on Npc A
- Analyzer finds that weight was < 0, reports topic, /w a `x => x.Weight` memberExpression
- File gets updated
- Engine checks if NpcA's weight field got changed
- It was!  Reruns the analyzer, finds that it's now 1, doesn't re-report topic

This is the "dream" of an optimized but live engine that is able to intelligently rerun analyzers only when appropriate.

So here, I think this is being misused, as the Armor's EditorID is not involved in determining if the analyzer should be rerun.  In fact... i think this analyzer is complex enough that it might not be able to specify a field to watch in this manner?  

I think we'll want to also upgrade the API to allow for "multiple members" to be watched easily, and maybe even a custom func comparer alternative.  And for the short term.. NO member as well (as right now it requires one to be specified)

https://github.com/Mutagen-Modding/Mutagen.Bethesda.Analyzers/issues/124